### ISSUE TITLE
If user clicks on Screenshots Library button without previously authenticating there's a server error

### ISSUE THREAD

**Issue by johngruen** [mozilla-services/screenshots#3439]
## STR
In a clean profile click the library button then choose Screenshots.

## Expected
I am navigated to My Shots

## Actual
I get a Server error page

Here's a video:
https://cl.ly/3e3y3G434510

**Comment by johngruen** [comment]
Actually i think the user should be navigated to the homepage

**Comment by ianb** [comment]
I can't reproduce this locally. I believe [this Sentry report](https://sentry.prod.mozaws.net/operations/pageshot-prod/issues/643488/) is the bug:

```
    shots = shots.map(function (shot) {return { id: shot.id, json: shot.asRecallJson() };});

/app/build/server/pages/shotindex/model.js in Object.exports.createModel at line 32:18
/app/build/server/reactrender.js in Object.exports.render at line 9:31
/app/build/server/pages/shotindex/server.js in _render at line 30:17
/app/build/server/pages/shotindex/server.js in null.<anonymous> at line 13:5
Called from: /app/node_modules/express/lib/router/layer.js in Layer.handle [as handle_request]
/app/build/server/middleware/l10n.js in null.<anonymous> at line 23:5
```

**Comment by ianb** [comment]
Ah, the exception is fixed in master in 75df15719432944970865af8d5ed781f4bfadd65

Note you still do get stuck on this page (infinite loop):

![image](https://user-images.githubusercontent.com/44390/29896527-f62a12f4-8da2-11e7-9170-d893b6150497.png)

But this is #2520 (I've bumped that ticket out to triage)
